protos/public_api.proto

```diff
@@ -381,10 +389,21 @@ message Permission {
     string slug = 3;
     string description = 4;
     string effect = 5;
-    repeated string resources = 6;
-    repeated string actions = 7;
+    repeated PermissionResource resources = 6;
+    repeated PermissionAction actions = 7;
 }
 
+message PermissionResource {
+    string id = 1;
+    string name = 2;
```

Comment:
maybe instead of `name` it should be called `bucket_name` and the pattern should also be separate, before a single string resource was representing both bucket and pattern, ie: `bucket-a/*` or `bucket-b/some/folder/key`, so if we are making the resources complete object that represent the database record maybe we could also split bucket_name and pattern

Reply:
done